docs/workspaces.md

```diff
@@ -72,6 +72,18 @@ specific `Volume` information to use for the `Workspaces` used by each `Pipeline
 `PipelineRuns` have the added responsibility of ensuring that whatever `Volume` type they
 provide can be safely and correctly shared across multiple `Tasks`.
 
+### Optional `Workspaces`
+
+Both Tasks and Pipelines can declare a Workspace "optional". When an optional Workspace
+is declared the TaskRun or PipelineRun may omit a Workspace Binding for that Workspace.
+The Task or Pipeline behaviour may change when the Binding is omitted. This feature has
+many uses:
+
+- A Task may optionally accept credentials to run authenticated commands.
+- A Pipeline may accept optional configuration that changes the linting or compilation
+parameters used.
+- An optional build cache may be provided to speed up compile times.
+
```

Comment:
I suggest that we add documentation about what happens when a workspace is optional but it has a default specified, saw that that case was covered in the tests but I'd imagine a user would wonder about that when reading the docs

https://github.com/tektoncd/pipeline/blob/dd548e6709be41db82e2e1f6e0574551530bd480/pkg/reconciler/taskrun/taskrun_test.go#L2851-L2855

Reply:
Great idea, I've updated the workspaces doc to add a new section which describes the default config in a bit more detail and includes this info as a note.